dimscord/dispatch.nim

```diff
@@ -939,84 +939,84 @@ proc voiceServerUpdate(s: Shard, data: JsonNode) {.async.} =
         asyncCheck s.client.events.interaction_create(s, newInteraction(data))
-    of "THREAD_CREATE": await s.threadCreate(data)
-    of "THREAD_UPDATE": await s.threadUpdate(data)
-    of "THREAD_DELETE": await s.threadDelete(data)
-    of "THREAD_LIST_SYNC":
+    of ThreadCreate: await s.threadCreate(data)
+    of ThreadUpdate: await s.threadUpdate(data)
+    of ThreadDelete: await s.threadDelete(data)
+    of ThreadListSync:
         asyncCheck s.client.events.thread_list_sync(s, ThreadListSync(
             channel_ids: data{"channel_ids"}.getElems.mapIt(it.getStr),
             threads: data{"threads"}.getElems.map(newGuildChannel),
             members: data["members"].elems.mapIt(it.`$`.fromJson(ThreadMember))
         ))
-    of "THREAD_MEMBERS_UPDATE": await s.threadMembersUpdate(data)
-    of "THREAD_MEMBER_UPDATE":
+    of ThreadMembersUpdate: await s.threadMembersUpdate(data)
+    of ThreadMemberUpdate:
         let guild = s.cache.guilds.getOrDefault(data["guild_id"].str,
             Guild(id: data["guild_id"].str)
         )
         asyncCheck s.client.events.thread_member_update(
             s, guild, data.`$`.fromJson(ThreadMember)
         )
-    of "STAGE_INSTANCE_CREATE": await s.stageInstanceCreate(data)
-    of "STAGE_INSTANCE_UPDATE": await s.stageInstanceUpdate(data)
-    of "STAGE_INSTANCE_DELETE": await s.stageInstanceDelete(data)
-    of "GUILD_SCHEDULED_EVENT_USER_ADD": await s.guildScheduledEventUserAdd data
-    of "GUILD_SCHEDULED_EVENT_USER_REMOVE": await s.guildScheduledEventUserRemove data
-    of "GUILD_SCHEDULED_EVENT_CREATE": await s.guildScheduledEventCreate data
-    of "GUILD_SCHEDULED_EVENT_UPDATE": await s.guildScheduledEventUpdate data
-    of "GUILD_SCHEDULED_EVENT_DELETE": await s.guildScheduledEventDelete data
-    of "AUTO_MODERATION_RULE_CREATE": await s.autoModerationRuleCreate data
-    of "AUTO_MODERATION_RULE_UPDATE": await s.autoModerationRuleUpdate data
-    of "AUTO_MODERATION_RULE_DELETE": await s.autoModerationRuleDelete data
-    of "AUTO_MODERATION_ACTION_EXECUTION":
+    of StageInstanceCreate: await s.stageInstanceCreate(data)
+    of StageInstanceUpdate: await s.stageInstanceUpdate(data)
+    of StageInstanceDelete: await s.stageInstanceDelete(data)
+    of GuildScheduledEventUserAdd: await s.guildScheduledEventUserAdd data
+    of GuildScheduledEventUserRemove: await s.guildScheduledEventUserRemove data
+    of GuildScheduledEventCreate: await s.guildScheduledEventCreate data
+    of GuildScheduledEventUpdate: await s.guildScheduledEventUpdate data
+    of GuildScheduledEventDelete: await s.guildScheduledEventDelete data
+    of AutoModerationRuleCreate: await s.autoModerationRuleCreate data
+    of AutoModerationRuleUpdate: await s.autoModerationRuleUpdate data
+    of AutoModerationRuleDelete: await s.autoModerationRuleDelete data
+    of AutoModerationActionExecution:
         await s.autoModerationActionExecution data
-    else:
-        discard
+    of Unknown: discard
```

Comment:
This is done instead of an `else` so that the case statement is `exhaustive`. Means we'll get a compile error if we add a new event but don't handle it